### PR TITLE
Add interactive login with token, GSR-151

### PR DESCRIPTION
**Pull request by genestack-solomatin**
Code of `_select` is moved without changes, but I have updated docstrings.

Shell example:

```
sa@pc> genestack-application-manager -H localhost:8080
   1) by token
   2) by email and password
   3) anonymously
How do you want to login: 1
Connecting to localhost:8080
token: 
genestack_client v0.22.0a1
Application manager shell.
Type 'help' for list of available commands.

tester@genestack.com>
```

```
sa@pc> genestack-application-manager -H localhost:8080
   1) by token
   2) by email and password
   3) anonymously
How do you want to login: 2
Connecting to localhost:8080
e-mail: tester@genestack.com
password for tester@genestack.com: 
genestack_client v0.22.0a1
Application manager shell.
Type 'help' for list of available commands.

tester@genestack.com> 
```
```
sa@pc > genestack-application-manager -H localhost:8080
   1) by token
   2) by email and password
   3) anonymously
How do you want to login: 3
genestack_client v0.22.0a1
Application manager shell.
Type 'help' for list of available commands.

anonymous>                   
```

### CODE REVIEW

File: genestack_client/utils.py [hunk 1]
```diff
@@ -137,6 +142,52 @@ def ask_confirmation(question, default=None):
         print 'Unexpected response please input "y[es]" or "n[o]"'
 
 
+def interactive_select(items, message, to_string=None, selected=None):
+    """
+    Asks user to choose one of the items.
+    This function will not end until user input correct answer.
+    This function does not check if the file is connected to a tty device.
+
```
Comment (genestack-solomatin) [inline genestack_client/utils.py]: I have updated docstring.

Code below is not changed, only moved.

**Comment by klas-genestack** [comment]
If this branch is still relevant, please fix conflicts and I'll review it (finally!)

**Comment by genestack-solomatin** [comment]
> If this branch is still relevant, please fix conflicts and I'll review it (finally!)

Yes, still relevant.   Conflicts resolved.

File: genestack_client/settings/genestack_user.py [hunk 2]
```diff
@@ -88,26 +88,43 @@ def __interactive_login(self, connection):
 
         email = self.email
         message = 'Connecting to %s' % self.host
+
+        choose_by_token = 'by token'
+        choose_by_email = 'by email and password'
+        choose_anonymously = 'anonymously'
```
Comment (klas-genestack) [inline genestack_client/settings/genestack_user.py]: `login_by_token, login_by_email, login_anonymously`
or even constant-like:
`LOGIN_BY_TOKEN, LOGIN_BY_EMAIL, LOGIN_ANONYMOUSLY`

File: genestack_client/settings/genestack_user.py [hunk 3]
```diff
@@ -88,26 +88,43 @@ def __interactive_login(self, connection):
...
             if message:
                 print(message)
-            if email and '@' in email:
-                email = raw_input('e-mail [%s]: ' % email).strip() or email
+
+            if choice == choose_by_email:
+                if email and '@' in email:
+                    email = raw_input('e-mail [%s]: ' % email).strip() or email
+                else:
+                    email = raw_input('e-mail: ').strip() or email
```
Comment (klas-genestack) [inline genestack_client/settings/genestack_user.py]: Generate prompt on condition, call `raw_input` afterwards once.

File: genestack_client/settings/genestack_user.py [hunk 4]
```diff
@@ -88,26 +88,43 @@ def __interactive_login(self, connection):
...
+                password = getpass('password for %s: ' % email)
+                try:
+                    connection.login(email, password)
+                    self.email = email
+                    self.password = password
+                    return
+                except GenestackAuthenticationException:
+                    message = ('Your username or password was incorrect for %s. '
```
Comment (klas-genestack) [inline genestack_client/settings/genestack_user.py]: `'Your username and password have been rejected by %s. '

Can we afford to change text message?  I hope so.

File: genestack_client/settings/genestack_user.py [hunk 5]
```diff
@@ -88,26 +88,43 @@ def __interactive_login(self, connection):
...
-            password = getpass('password for %s: ' % email)
-            try:
-                connection.login(email, password)
-                self.email = email
-                self.password = password
-                return
-            except GenestackAuthenticationException:
-                message = 'Your username or password was incorrect for %s. Please try again.' % self.host
+                except GenestackAuthenticationException:
+                    message = 'Your token was incorrect for %s. Please try again.' % self.host
```
Comment (klas-genestack) [inline genestack_client/settings/genestack_user.py]: 'Your token has been rejected by %s. Please try again.'

File: genestack_client/utils.py [hunk 6]
```diff
@@ -121,7 +121,8 @@ def get_connection(args=None):
 def ask_confirmation(question, default=None):
     """
     Ask confirmation and return response as boolean value.
-    This method will not end until user input correct answer.
+    This function will not end until user input correct answer.
+    This function does not check if the file is connected to a tty device.
```
Comment (klas-genestack) [inline genestack_client/utils.py]:
```
Will be looping until user provides correct credentials.
Does not check if program is attached to tty device.
```

File: genestack_client/utils.py (hunk 1 above)
Comment (klas-genestack) [reply genestack_client/utils.py]: The same as above
```
Will be looping until user provides correct credentials.
Does not check if program is attached to tty device.
```

**Pull request merged**